### DISCUSSION TITLE
Combine images into one image, and also it doesn't work

### DISCUSSION THREAD

**Discussion by RMcD94** [ImageMagick/ImageMagick#2582] [Help]
I downloaded the program. When I type in the thing it wants into command prompt it says unrecognized. When I click the exe nothing happens.

I don't know how to program. I've been trying to combine a bunch of images into a single image. Honestly it would have been quicker to do it manually in paint but there's 256 images.

Also the images are in the wrong order so the columns should be rows and vice versa and I don't know how to fix that 

Is there a discord? I feel like someone capable could have done this in 30s and the final image is only like 2 MB

![image](https://user-images.githubusercontent.com/13921691/93267253-49e9a400-f7a3-11ea-8a1e-9258f15ac920.png)

**Comment by tml** [comment]
Well, I hope you're adhering to [OpenStreetMap's rules for use of tiles](https://operations.osmfoundation.org/policies/tiles/); however, imagemagick isn't a tool with a traditional Graphical User Interface ("GUI") that you can open and interact with. It uses instead a Command-Line Interface ("CLI"), where you have to pass arguments to the command-line telling ImageMagick what to do. There's plenty of documentation online, for example at https://imagemagick.org/script/command-line-processing.php, but you are probably looking for something like this (note: I am using Linux, not Windows; I do not use Windows, so I can't really help you much with how you might need to change these commands to make them work on Windows):

```
for y in {0..5}; do for x in {0..9}; do curl -fsk  -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:80.0) Gecko/20100101 Firefox/80.0'  "https://a.tile.openstreetmap.org/4/${y}/${x}.png" -o "4${y}${x}.png"; done; done # download the tiles

# Stitch all the tiles in a given "column" together - 
# I'm assuming some things about the tiles here,
# which may not be correct, so you'll have to test
# these results for yourself and adjust it if my
# assumptions prove wrong.

convert 40*.png -append c400.png
convert 41*.png -append c410.png
convert 42*.png -append c420.png
convert 43*.png -append c430.png
convert 44*.png -append c440.png
convert 45*.png -append c450.png

# Now stitch all the "columns" together
convert c*.png +append map.png
```

You can read more about the "append" flag here: https://imagemagick.org/script/command-line-options.php#append

At least for me, this resulted in something that looked likely:
![map](https://user-images.githubusercontent.com/35148/93271630-fe1e0700-f770-11ea-9f5e-e7594ab53f0d.png)
